internal/state/api/members.go

```diff
@@ -176,3 +202,83 @@ func (s *Server) getUser(w http.ResponseWriter, r *http.Request, p httprouter.Pa
 
 	return s.writeTerm(w, m)
 }
+
+func (s *Server) getUsers(w http.ResponseWriter, r *http.Request, p httprouter.Params) error {
+	// id1,id2,id3
+	userIDs := r.URL.Query().Get("userIDs")
+	
+	if userIDs == "" {
+		return xerrors.Errorf("empty userids array")
+	}
+
+	userIDsSlice := strings.Split(userIDs, ",")
+
+	var convertedUserIDs []int64
+
+	for _, userId := range userIDsSlice {
+		value, err := strconv.ParseInt(userId,10,64);
+		if err != nil {
+			return xerrors.Errorf("parse int for userids array: %w", err)
+		}
+		convertedUserIDs = append(convertedUserIDs,value)
+	}
+
+	data, err := s.db.GetUsersDiscordIdAndUsername(r.Context(), convertedUserIDs)
+	if err != nil {
+		return xerrors.Errorf("read users: %w", err)
+	}
+
+	result := make([][]byte, len(data))
+
+    for i, userAndData := range data {
+        // Convert each UserAndData to []byte
+        userAndDataBytes, err := json.Marshal(userAndData)
+        if err != nil {
+            return xerrors.Errorf("converting user ids and discord data to [][]byte: %w", err)
+        }
+
+        result[i] = userAndDataBytes
+    }
+
+	return s.writeTerms(w, result)
+}
+
+// Define a handler function for SetGuildMembers
+func (s *Server) setGuildMembers(w http.ResponseWriter, r *http.Request, p httprouter.Params) error {
+    // Parse guild ID from the URL parameter
+	guild, err := guildParam(p)
+	if err != nil {
+		return xerrors.Errorf("read guild param: %w", err)
+	}
+	
+	// Seems dumb to read into [string]GuildMember first before converting to [int64][]byte
```

Comment:
can try to directly convert to `[string][]byte`?

Reply:
I tried previously and seemed to mess up which is why I started using this GuildMember struct. I will try again though.